manifests/syslinux/system.pp

```diff
@@ -8,24 +8,19 @@
     source  => "${syslinux_dir}/pxelinux.0",
     require => Package['syslinux'],
   }
-
-  file { "${tftp_root}/syslinux/menu.c32":
-    source  => "${syslinux_dir}/menu.c32",
-    require => Package['syslinux'],
-  }
-
-  file { "${tftp_root}/syslinux/vesamenu.c32":
-    source  => "${syslinux_dir}/vesamenu.c32",
-    require => Package['syslinux'],
-  }
-
-  file { "${tftp_root}/syslinux/reboot.c32":
-    source  => "${syslinux_dir}/reboot.c32",
-    require => Package['syslinux'],
-  }
-
-  file { "${tftp_root}/syslinux/memdisk":
-    source  => "${syslinux_dir}/memdisk",
-    require => Package['syslinux'],
+  $tftproot_syslinux_files = [
+    'menu.c32',
+    'vesamenu.c32',
+    'reboot.c32',
+    'ldlinux.c32',
+    'libcom32.c32',
+    'libutil.c32',
+    'memdisk',
+  ]
+  $tftproot_syslinux_files.each |$file| {
+    file { "${tftp_root}/syslinux/${file}":
+      source  => "${syslinux_dir}/${file}",
+      require => Package['syslinux'],
+    }
```

Comment:
Maybe we can manage the directory itself.  If `source` is a directory and `recurse` is set to true, it should copy all files and not need an update if the files change at some point.

```suggestion
    file { "${tftp_root}/syslinux":
      source  => $syslinux_dir,
      recurse => true,
      require => Package['syslinux'],
    }
```

Reply:
Made the changes I thought would work, but I'm getting CI errors on BSD. An extra set of eyes would be appreciated.